### PR TITLE
Implement render for kustomize

### PR DESCRIPTION
**Pull request by arminbuerkle**
Relates to #1937 and #2957 for kustomize.

**Description**

Implements the `render` command for kustomize. Basically copied the way `kubectl render` works to kustomize to try it out. It works for my current purpose so i added a unit test.

If it looks good you can merge it, if someone else is already working on this feel free to close it.

**User facing changes**
Implements the `render` command for kustomize.

**Next PRs.**

n/a

**Submitter Checklist**

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [unit tests](../DEVELOPMENT.md#creating-a-pr)
- [x] Mentions any output changes.
- [x] Adds documentation as needed: user docs, YAML reference, CLI reference. (Not needed?)
- [x] Adds integration tests if needed. (Not needed?)

**Reviewer Notes**

- [ ] The code flow looks good. 
- [ ] Unit test added.
- [ ] User facing changes look good.

**Release Notes**

```
- Implement `skaffold render` for `kustomize` deployer
```

### CODE REVIEW

File: pkg/skaffold/deploy/kustomize.go [hunk 1]
```diff
@@ -121,33 +117,48 @@ func (k *KustomizeDeployer) Deploy(ctx context.Context, out io.Writer, builds []
...
+	if err := k.kubectl.Apply(ctx, textio.NewPrefixWriter(out, " - "), manifests); err != nil {
+		event.DeployFailed(err)
+		return NewDeployErrorResult(errors.Wrap(err, "kubectl error"))
+	}
+
+	event.DeployComplete()
+	return NewDeploySuccessResult(namespaces)
+}
+
+func (k *KustomizeDeployer) renderManifests(ctx context.Context, out io.Writer, builds []build.Artifact, labellers []Labeller) (deploy.ManifestList, error) {
```
Comment (arminbuerkle) [inline pkg/skaffold/deploy/kustomize.go]: I'm not sure why `Render` doesn't have `[]Labeller` as parameter. I think that's something that should get added in the future so the behaviour is similar to `Deploy`.

**Comment by arminbuerkle** [comment]
@dgageot No sure whom to mention but is this PR going to make it? Should i keep it updated?

**Comment by dgageot** [comment]
Sorry @arminbuerkle, the team was quite busy launching version 1.0. I guess it's best to ping @nkubala

**Comment by arminbuerkle** [comment]
@nkubala friendly reminder that having this feature would be awesome and improve the gitops workflow with kustomize mentioned in #1937 & #2957

File: pkg/skaffold/deploy/kustomize.go (hunk 1 above)
Comment (nkubala) [reply pkg/skaffold/deploy/kustomize.go]: it should have it. looks like maybe I forgot to put it in there when I added the boilerplate a while back.

File: pkg/skaffold/deploy/kustomize.go [hunk 2]
```diff
@@ -97,15 +98,10 @@ func (k *KustomizeDeployer) Labels() map[string]string {
 
 // Deploy runs `kubectl apply` on the manifest generated by kustomize.
 func (k *KustomizeDeployer) Deploy(ctx context.Context, out io.Writer, builds []build.Artifact, labellers []Labeller) *Result {
-	if err := k.kubectl.CheckVersion(ctx); err != nil {
-		color.Default.Fprintln(out, "kubectl client version:", k.kubectl.Version(ctx))
-		color.Default.Fprintln(out, err)
-	}
-
-	manifests, err := k.readManifests(ctx)
+	manifests, err := k.renderManifests(ctx, out, builds, labellers)
```
Comment (nkubala) [inline pkg/skaffold/deploy/kustomize.go]: I think there should be a `event.DeployInProgress()` here.

Comment (arminbuerkle) [reply pkg/skaffold/deploy/kustomize.go]: You're right. Added!

**Comment by arminbuerkle** [comment]
@nkubala PTAL. I fixed the event handling. 

The labeling requires changing the `Render` interface and consequently all of its implementations. Not really something that should be in this PR.

I'm happy to submit a follow up PR that fixes labeling and introduces a cmd flag to allow setting labels when using skaffold render.

**Comment by alnr** [comment]
Is there any update on this PR?

**Pull request merged**